src/uint256.h

```diff
@@ -112,21 +111,11 @@ class uint256 : public base_blob<256> {
     static const uint256 ONE;
 };
 
-/* uint256 from const char *.
- * This is a separate function because the constructor uint256(const char*) can result
- * in dangerously catching uint256(0).
+/* uint256 from std::string_view.
+ * This is not a uint256 constructor because of historical fears of uint256(0)
```

Comment:
nit: I interpret "historical fears" as "the fears are no longer applicable", in which case I think it should just be a constructor and this helper function removed. But, it seems like until C++23 this is indeed an issue (even if the ambiguous overload with the `uint8_t` should prevent compilation), so the current wording is confusing to me.

Suggested alternative:

```
/* uint256 from std::string_view.
 * This is not a uint256 constructor to avoid the ambiguity of
 * uint256(0) being interpreted as initialization from a nullptr.
 *
 * This helper can be replaced with a constructor when we require c++23,
 * which prohibits constructing a string_view from a nullptr, see
 * https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2166r1.html
 */
```

Reply:
Thank you for the correction! Applied with some minor adjustments.

Reply:
Reset back to prior version, see https://github.com/bitcoin/bitcoin/pull/30436#discussion_r1682274787.